lib/mocha.js

```diff
@@ -914,14 +930,14 @@ Mocha.prototype.run = function(fn) {
     this._previousRunner.dispose();
     this.suite.reset();
   }
-  if (this.files.length && !this.loadAsync) {
+  if (this.files.length && !this.lazyLoadFiles) {
```

Comment:
`lazyLoadFiles` was originally `loadAsync`, and that was used by the ESM-loading logic, because `async import()` was used.  So, it's not unique to parallel mode--but I'm open to ideas about how best to handle this.

Reply:
It's fine for now. We can always refactor it later. Should have been `_lazyLoadFiles` instead of `lazyLoadFiles` but I see you caught it already in your latest push :)